### PR TITLE
VOs resources page

### LINKED ISSUE #22: Vo resources page
* The goal of this task is to create a tab for resources displayed in vo detail.
* By default, there should be displayed all resources and it should be possible to filter them by name. 
* To get the resources, you need the `rpc/jsonp/resourcesManager/getRichResources` method.
* You also need to create a resources service.
* In the displayed table, there should be columns: `checkbox`, `resourceId`, `name`, `facility`, `tags` and `description`.
* There should also be a delete button, but you don't need to implement it in this issue. It can be done separately.

### PR DESCRIPTION
**Pull request by DanoFecko**
Resolves #22

### CODE REVIEW

**Comment by DanoFecko** [comment]
I changed check boxes in table to ones in shown here https://material.angular.io/components/table/overview

File: src/app/vos/vo-detail-page/vo-resources/vo-resources.component.html [hunk 1]
```diff
@@ -1,3 +1,12 @@
...
+  <h1 class="page-subtitle">{{'VO_DETAIL.RESOURCES.TITLE' | translate}}</h1>
+
+    <button mat-flat-button color="warn" class="ml-2" [disabled]="selectedResources.size === 0">
+      {{'VO_DETAIL.RESOURCES.DELETE' | translate}}
+    </button>
+
+      <app-resources-list
+        [resources]="resources"
+        (resourceSelectChange)="onResourceSelectChange($event)">
+      </app-resources-list>
```
Comment (Vojtech-Sassmann) [inline src/app/vos/vo-detail-page/vo-resources/vo-resources.component.html]: Please fix indentation in this file.

File: src/app/vos/resources-list/resources-list.component.html [hunk 2]
```diff
@@ -0,0 +1,64 @@
+<mat-form-field class="ml-3">
+  <input matInput (keyup)="applyFilter($event.target.value)"
+         placeholder="{{'VO_DETAIL.RESOURCES.TABLE_SEARCH' | translate}}">
+</mat-form-field>
+<div class="row">
```
Comment (Vojtech-Sassmann) [inline src/app/vos/resources-list/resources-list.component.html]: Please, remove this div. It is unnecessary and it breaks the alignment of items.

File: src/app/vos/vo-detail-page/vo-resources/vo-resources.component.html [hunk 3]
```diff
@@ -1,3 +1,12 @@
-<p>
-  vo-resources works!
-</p>
+<div>
+  <h1 class="page-subtitle">{{'VO_DETAIL.RESOURCES.TITLE' | translate}}</h1>
+
+    <button mat-flat-button color="warn" class="ml-2" [disabled]="selectedResources.size === 0">
```
Comment (Vojtech-Sassmann) [inline src/app/vos/vo-detail-page/vo-resources/vo-resources.component.html]:
```suggestion
    <button mat-flat-button color="warn" [disabled]="selectedResources.size === 0">
```

File: src/app/vos/resources-list/resources-list.component.ts [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+  @ViewChild(MatSort) set matSort(ms: MatSort) {
+    this.sort = ms;
+    this.setDataSource();
+  }
+
+  @Input()
+  resources: RichResource[] = [];
+
+  @Output()
+  resourceSelectChange: EventEmitter<ResourceSelectChange> = new EventEmitter<ResourceSelectChange>();
```
Comment (Vojtech-Sassmann) [inline src/app/vos/resources-list/resources-list.component.ts]: This event is never emitted, so the selection of resources is not working. Is it left out on purpose?

File: src/assets/i18n/en.json [hunk 5]
```diff
@@ -33,6 +33,16 @@
...
+    },
+    "RESOURCES": {
+      "TITLE": "Resources",
+      "DELETE": "Remove selected",
+      "TABLE_RESOURCE_ID": "id",
+      "TABLE_RESOURCE_NAME": "Name",
+      "TABLE_FACILITY_NAME": "Facility name",
+      "TABLE_RESOURCE_TAGS": "Tags",
+      "TABLE_RESOURCE_DESCRIPTION": "Description",
+      "TABLE_SEARCH": "Search"
```
Comment (Vojtech-Sassmann) [inline src/assets/i18n/en.json]:
```suggestion
      "TABLE_SEARCH": "Filter by name or description"
```

Also, if the text doesn't fit, you can make the input a bit wider.

**Comment by Vojtech-Sassmann** [comment]
And also, next time add the PR to the project.

**Comment by DanoFecko** [comment]
I deleted unused code, and changed color of checkboxes @Vojtech-Sassmann

File: src/app/vos/resources-list/resources-list.component.html [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+      </ng-container>
+      <ng-container matColumnDef="facility">
+        <th mat-header-cell *matHeaderCellDef
+            mat-sort-header>{{'VO_DETAIL.RESOURCES.TABLE_FACILITY_NAME' | translate}}</th>
+        <td mat-cell *matCellDef="let resource">{{resource.facility.name}}</td>
+      </ng-container>
+      <ng-container matColumnDef="tags">
+        <th mat-header-cell *matHeaderCellDef
+            mat-sort-header>{{'VO_DETAIL.RESOURCES.TABLE_RESOURCE_TAGS' | translate}}</th>
+        <td mat-cell *matCellDef="let resource">{{resource.resourceTags}}</td>
```
Comment (Vojtech-Sassmann) [inline src/app/vos/resources-list/resources-list.component.html]: This doesn't display tags. It just displays `[Object object]`. I suggest creating a pipe that would transform a field of tags into a string.

**Comment by DanoFecko** [comment]
I added pipe for "translation" of resource tags to string

**Comment by DanoFecko** [comment]
I added pipe for "translation" of resource tags to string

**Pull request merged**